packages/website/docs/sources.md

```diff
@@ -82,25 +58,87 @@ const autocomplete = createAutocomplete({
         getItemUrl({ item }) {
           return item.url;
         },
+        // ...
       },
     ];
   },
 });
 ```
 
-To bring more search capabilities, you can plug an Algolia index:
+Before moving on to more complex sources, let's take a closer look at the code.
+
+Notice that the [`getSources`](#getsources) function returns an array of sources. Each source implements a [`getItems`](#getitems) function to return the items to display. These items can be a simple static array, but you can also use a function to refine items based on the query. **The [`getItems`](#getitems) function is called whenever the input changes.**
+
+By default, autocomplete items are meant to be hyperlinks. To determine what URL to navigate to, you can implement a [`getItemURL`](#getitemurl) function. It enables the [keyboard accessibility](https://autocomplete.algolia.com/docs/keyboard-navigation) feature, allowing users to open items in the current tab, a new tab, or a new window from their keyboard.
+
+### Customizing items with templates
 
-```ts
+In addition to defining data sources for items, a source also lets you customize how to display items using [`templates`](#templates).
+
+```js
+autocomplete({
+  // ...
+  getSources({ query }) {
+    return [
+      {
+        // ...
+        templates: {
+          item({ item }) {
+            return `Result: ${item.name}`;
+          },
+        },
+      },
+    ];
+  },
+});
+```
+
+You can also display header and a footer elements around the list of items.
+
+```js
+autocomplete({
+  // ...
+  getSources({ query }) {
+    return [
+      {
+        // ...
+        templates: {
+          header() {
+            return 'Suggestions';
+          },
+          item({ item }) {
+            return `Result: ${item.name}`;
+          },
+          footer() {
+            return 'Footer';
+          },
+        },
+      },
+    ];
+  },
+});
+```
+
+**Templates aren't limited to strings.** You can provide anything as long as they're a valid virtual DOM element (see more in [**Templates**](templates)).
```

Comment:
The link to **Templates** fails the build. Can we add the link in the other PR?

Reply:
Done in e0d1748.